infra/presubmit.py

```diff
@@ -132,22 +132,32 @@ def check_project_yaml_constants(self):
         continue
       actual_constants = self.data[section]
       for constant in actual_constants:
-        if constant not in allowed_constants:
-          self.print_error('%s (in %s section) is not one of %s', constant,
-                           section, allowed_constants)
+        if isinstance(constant, str):
+          if constant not in allowed_constants:
+            self.error(
+                '{constant} (in {section} section) is not one of {allowed}.'.
+                format(constant=constant,
+                       section=section,
+                       allowed=allowed_constants))
+        else:
+          # The only alternative value allowed is the experimental flag, i.e.
+          # `constant == {'memory': None, 'experimental': True}`.
```

Comment:
I am confused here, will need to check in person.

Reply:
This is how 

```yaml
sanitizers:
 - address
 - memory:
    experimental: True
 - undefined
 - dataflow
```

gets parsed, everything except for `memory` are just strings, and for the `memory` one it's a dict like the one in the comment

Reply:
Isn't the format here:

```python
{
  'memory': {
    'experimental': True
  },
}
```
Rather than the flat format described in the comment?

Reply:
So this one does not test if like memory is in allowed_constants. that feels more important than checking if "experimental" child is there (can be even skipped for complexity).
if isinstance(constant, dict):
 constant = list(constant.keys())[0]
before the 
if constant not in allowed_constants:

@oliverchang - thoughts ?

Reply:
```
>>> yaml.load('''
... sanitizers:
 - address
 - memory:
    experimental: True
 - undefined
 - dataflow''')
__main__:2: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
{'sanitizers': ['address', {'memory': {'experimental': True}}, 'undefined', 'dataflow']}
>>> 
```

Which means the correct check will involve:
- Check that the dict has one key/value pair.
- Check that the value is a dict with one key/value pair and the only key is 'experimental'.

Reply:
@inferno-chromium yeah agreed we can probably just skip the experimental check to keep things simple, since that is not common and mostly done by us.

Reply:
I also was expecting the dict you specified, but apparently it's not the case:

```
$ python3
Python 3.7.5rc1 (default, Dec 20 2019, 17:52:56) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import yaml
>>> y = yaml.safe_load('projects/wolfssl/project.yaml')
>>> y
'projects/wolfssl/project.yaml'
>>> y = yaml.safe_load(open('projects/wolfssl/project.yaml'))
>>> print(y['sanitizers'])
['address', {'memory': None, 'experimental': True}, 'undefined', 'dataflow']
```

Reply:
looks like the indentation matters in those yaml files, if I add one more space before `experimental: True` it looks right. So this one I broken, but there is also `rapidjson.yaml` affected by it -- will fix.